pkg/infra/server/baseServer.go

```diff
@@ -40,19 +27,60 @@ func (bs *BaseServer) RunGameLoop(loopnum int) {
 		fmt.Printf("agent %d  \n", index)
 		fmt.Printf("_____________________________________________ \n")
 		agent.UpdateAgent()
-
+		agent.Activity()
 		//TO DO: add the function for stages
 
 	}
 
 }
 
-func (bs *BaseServer) Start() {
-	bs.Init()
+func (bs *BaseServer) Start(m []AgentGeneratorCountPair) {
+	bs.Init(m)
 	//LOOPS
 	for i := 0; i < bs.NumTurns; i++ {
 		fmt.Printf("Loop: %d \n", i)
 		bs.RunGameLoop(i)
 	}
 
 }
```

Comment:
the init shouldn't be done here. It's weird to have two functions that accept the map